R/addAlpha.R

```diff
@@ -0,0 +1,229 @@
+#' Estimate alpha diversity indices.
+#' 
+#' The function estimates alpha diversity indices optionally using of rarefaction,
+#' then stores results at \code{\link{colData}}.
+#' 
+#' @param x a \code{\link{SummarizedExperiment}} object.
+#' 
+#' @param assay.type the name of the assay used for
+#'   calculation of the sample-wise estimates (default: \code{assay.type = "counts"}).
+#'   
+#' @param index a \code{character} vector, specifying the alpha diversity indices
+#'   to be calculated.
+#'   
+#' @param name a name for the column(s) of the colData the results should be
+#'   stored in. By default this will use the original names of the calculated
+#'   indices(By default: \code{name = index}).
+#'   
+#' @param ... optional arguments.
+#' 
+#' @param n.iter a single \code{integer} value for the number of rarefaction
+#' rounds (By default: \code{n.iter = 10}).
+#' 
+#' @param rarefaction.depth a \code{double} value as for the minimim size or 
+#' rarefaction.depth. (By default: \code{rarefaction.depth = NULL})
+#' 
+#' @return \code{x} with additional \code{\link{colData}} named after the index 
+#' used.
+#' 
+#' @examples
+#' 
+#' data("GlobalPatterns")
+#' tse <- GlobalPatterns
+#' 
+#' # Calculate the default Shannon index with no rarefaction
+#' tse <- addAlpha(tse, assay.type = "counts", index = "shannon")
+#' 
+#' # Shows the estimated Shannon index
+#' tse$shannon
+#'
+#' # Calculate observed richness with 10 rarefaction rounds
+#' tse <- addAlpha(tse, assay.type = "counts", index = "observed_richness",
+#' rarefaction.depth=min(colSums(assay(tse, "counts")), na.rm = TRUE), n.iter=10)
+#' 
+#' # Shows the estimated observed richness
+#' tse$observed_richness
+#' 
+
+#' @name addAlpha
+#' @rdname addAlpha
+#' @export
+setGeneric(
+    "addAlpha", signature = c("x"), function(
+        x, assay.type = "counts", 
```

Comment:
Check: is this PR suggesting an entirely new function "addAlpha", or is this modification of the already existing function "addAlpha" that did not have rarefaction before?

Ideally, this PR will extend the existing alpha diversity functionality and only add the parts that are needed to add rarefaction.

Reply:
This is PR is about to add the `addAlpha` function and also depricate and change previous estimate*Alpha* functions.

Reply:
Perfect.